### PR TITLE
[ENH] Featurizer simplified

### PR DESCRIPTION
**Pull request by fkiraly**
This simplifies @aiwalter's `Featurizer` from #1727 as follows.

Essentially, @aiwalter's `Featurizer` was a pipeline composite of three things: (a) switching `y` to `X`, (b) applying the `transformer`, (c) applying a lag.

This seems unnecessarily complicated, since the same effect can be achieved by (a), and a transformer and finally `Lag`, chained in a `TransformerPipeline`.

In the example, `Featurizer() * ExponentTransformer() * Lag(lags)` instead of `Featurizer(lags=lags, transformer=ExponentTransformer())`, which is more design parsimonous and python zen.

So, we can simply create a transformer for (a) and achieve the rest by pipeline composition.

Added benefit is having everything for hierarchical data, and the ability to specify `lags` in temporal units, not just ints (as in the `Lag` transformer).

What do you think, @aiwalter? Still the same idea.

I have changed your test, `test_featurized_values`, to test the above pipeline.

Further additions/changes in this PR:
* name changed to `YtoX`, upon suggestion of @aiwalter
* transformer scenarios with `y` were sharpened: scenario with `Series` `y`, as needed for `Featurizer`, was added. `is_applicable` cleaned up to match scenarios with `y` only with estimators suitable for that specific scitype of `y`.

### CODE REVIEW

**Comment by fkiraly** [comment]
tests fail because the scenario passed has classifier labels `y` and not series `y`. This needs to be made more granular. Will fix this is @aiwalter is happy with the idea.

**Review by aiwalter — COMMENTED** [review]
Should we rename the class or keep `Featurizer `? Could be `YtoX`? Also, how can this work if `X` is multivariate but I only want to apply a univariate transformer on on of the columns to be added as new feature? I am a bit confused about swithcing `X` and `y`, in the end my original idea was to extend `y` with new column(s) created from `X`. So this is not switching actually so I am not sure if this PR is actually doing what I was intended to do

**Comment by fkiraly** [comment]
> I am a bit confused about swithcing `X` and `y`, in the end my original idea was to extend `y` with new column(s) created from `X`. So this is not switching actually so I am not sure if this PR is actually doing what I was intended to do

I see - there are two ways to *add* the features instead of replacing them:

* using `FeatureUnion`, e.g., 
`Lag() + Featurizer() * ExponentTransformer() * Lag(lags)`
* we could also add an argument, say, `behaviour` which can be `replace` or `union`, where we column append `y`

File: sktime/transformations/series/compose.py [hunk 1]
```diff
@@ -429,3 +429,63 @@ def _check_is_pdseries(z):
...
+        "scitype:y": "both",
+        "fit_is_empty": True,
+        "requires_y": True,
+    }
+
+    def __init__(self):
+        super(Featurizer, self).__init__()
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply sktime/transformations/series/compose.py:457]: this is the standard blurb in the docstring, but, sure.

**Comment by fkiraly** [comment]
>Should we rename the class or keep `Featurizer `? Could be `YtoX`?

Yes, `YtoX` is probably better.

File: sktime/transformations/series/compose.py [hunk 2]
```diff
@@ -429,3 +429,63 @@ def _check_is_pdseries(z):
...
+
+        core logic
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply sktime/transformations/series/compose.py]: replaced by the default blurb

**Pull request merged**